src/panel/context/Explorer/ast/index.ts

```diff
@@ -1,177 +1,221 @@
 import stringify from "fast-json-stable-stringify";
 import nanoid from "nanoid";
-import { Operation } from "urql";
-
-import { Scalar, SelectionSet, Variables, Context, NullArray } from "./types";
-
+import { Operation, OperationDebugMeta, OperationResult } from "urql";
 import {
-  getName,
-  getSelectionSet,
-  isFieldNode,
-  isInlineFragment,
-  getFieldAlias
-} from "./node";
-
-import { getFieldArguments, normalizeVariables } from "./variables";
-
-import { getMainOperation, getFragments } from "./traversal";
-
-type DataField = Scalar | NullArray<Scalar> | null;
-
-export interface FieldNode {
+  SelectionNode,
+  Kind,
+  FieldNode,
+  InlineFragmentNode,
+  FragmentSpreadNode,
+  OperationDefinitionNode,
+  FragmentDefinitionNode,
+  ASTNode
+} from "graphql";
+import { getFieldArguments, getNormalizedVariables } from "./variables";
+
+export interface ParsedFieldNode {
   _id: string;
   _owner: {};
-  cacheOutcome?: Context["cacheOutcome"];
+  cacheOutcome?: OperationDebugMeta["cacheOutcome"];
   key: string;
   name: string;
-  args?: Variables | null;
-  value?: DataField | null;
-  children?: NodeMap | NullArray<NodeMap>;
+  args?: Operation["variables"];
+  value?: OperationResult["data"];
+  children?: ParsedNodeMap | (ParsedNodeMap | null)[];
 }
 
-export type NodeMap = Record<string, FieldNode>;
+export type ParsedNodeMap = Record<string, ParsedFieldNode>;
 
-interface Data {
-  [fieldName: string]: Data[] | Data | DataField;
+interface HandleResponseArgs {
+  operation: Operation;
+  data: OperationResult["data"];
+  parsedNodes?: ParsedNodeMap;
 }
 
-export const keyOfField = (fieldName: string, args?: null | Variables) =>
-  args ? `${fieldName}(${stringify(args)})` : fieldName;
-
-export const startQuery = (
-  request: Operation,
-  data: Data,
-  map: NodeMap = Object.create(null)
-) => {
-  if (request.operationName !== "query") {
-    return map;
+export const handleResponse = ({
+  operation,
+  data,
+  parsedNodes = {}
```

Comment:
I'm assuming we're not so concerned with bundlesize here in `devtools`? Also can see the logic in sticking with undefined consistently if using default params 👍

Reply:
Yeah I'm not too concerned about bundle size as assets are locally stored when installing the extension